Fixes after review #1

Chantreck opened the pull request.

Commit:
Optimized drawing code #1

1. Changed status bar color to white and decreased number of 8s in hint
2. Moved duplicated code to functions
3. Moved string constants to strings.xml
4. Moved repeated constants and settings to variables (in ui.theme)
5. Moved buttons labels to enum but not implemented new logic yet

Commit:
Moved business logic to model layer

Commit:
Moved strings to constants

Commit:
UI fixes

ashokolov commented on app/src/main/java/com/tsu/calculator/MainModel.kt:
```diff
@@ -1,35 +1,185 @@
+
+    private var _firstOperand: String = ""
+    private var _secondOperand: String = ""
+    private var _operation: Buttons? = null
```
In kotlin we use symbol _ in private variables only if it have val getter. It's not critical but better to remove it
example: 

> private var _number: Int = 0
> val number: Int
>     get() = _number

ashokolov commented on app/src/main/java/com/tsu/calculator/MainModel.kt:
```diff
@@ -1,35 +1,185 @@
 
         when (operator) {
-            "divide" -> {
+            Buttons.DIVIDE -> {
```
Great! <3

ashokolov commented on app/src/main/java/com/tsu/calculator/MainModel.kt:
```diff
@@ -1,35 +1,185 @@
+    private var _secondOperand: String = ""
+    private var _operation: Buttons? = null
+
+    companion object {
```
Usually inner classes locates in top of class or on bottom, before of after all variables and functions

ashokolov commented on app/src/main/java/com/tsu/calculator/MainModel.kt:
```diff
@@ -1,35 +1,185 @@
             }
             else -> {
-                throw Exception("Unknown operator")
+                Log.e("Model", OPERATOR_ERROR + " " + operator.label)
```
Good practice to use `Timber` instead of `Log`

ashokolov commented on app/src/main/java/com/tsu/calculator/MainModel.kt:
```diff
@@ -1,35 +1,185 @@
     }
+
+    fun getResult(): String {
+        val operation = _operation
```
Copy this variable is unnecessary

ashokolov commented on app/src/main/java/com/tsu/calculator/MainModel.kt:
```diff
@@ -1,35 +1,185 @@
+
+    fun setOperation(operation: Buttons): String {
+        if (_firstOperand.isEmpty() && _secondOperand.isEmpty()) {
+            _firstOperand = "0"
```
Move "0" to const variable inside this class

ashokolov commented on app/src/main/java/com/tsu/calculator/MainModel.kt:
```diff
@@ -1,35 +1,185 @@
+        }
+
+        if (_secondOperand.isEmpty() || _secondOperand == "0") {
+            if (_firstOperand[0] == '-') {
```
Poor practice to compare strings. You should use enum classes instead

For example your enum class could contain an operation string (such you did in in `Buttons` enum). Then if you need to get string value from `enum` you could call `Buttons.label`. If you need to get `enum class` from string you could call `Buttons.valueOf(string)`

If this is only unary minus you could declare const variable like `unaryMinus = "-"` and compare with this (but compare strings still isn't so good and it's better to use enums)

ashokolov commented on app/src/main/java/com/tsu/calculator/MainModel.kt:
```diff
@@ -1,35 +1,185 @@
+            if (commaAdded) {
+                return ""
+            }
+            _secondOperand += "."
```
Move "." to const variable

ashokolov commented on app/src/main/java/com/tsu/calculator/MainViewModel.kt:
```diff
@@ -21,154 +14,57 @@ class MainViewModel : ViewModel() {
-                _firstOperand
+            Buttons.DIVIDE, Buttons.MULTIPLY, Buttons.MINUS, Buttons.PLUS -> {
+                var temp = model.setOperation(button)
+                if (temp.endsWith(".0")) {
```
Move ".0" to const variable or enum

ashokolov replied:
Still here

Commit:
Fixed some remarks

1. Deleted _ before private variables in model
2. Moved companion object in model to the top of the class
3. Changed Log to Timber
4. Deleted operation copying in getResult() of model
5. Moved some strings to constants (planning to get rid of some of them futher)

Commit:
Test commit

Commit:
Changed changeSign method in model

ashokolov commented on app/src/main/java/com/tsu/calculator/MainModel.kt:
```diff
@@ -1,35 +1,187 @@
+            }
+        }
+
+        val result = number.toDouble() / 100.0
```
Magic number. Put it into variable

ashokolov commented on app/src/main/java/com/tsu/calculator/MainViewModel.kt:
```diff
@@ -21,154 +14,57 @@ class MainViewModel : ViewModel() {
+        when (button) {
+            Buttons.CLEAR -> {
+                model.clear()
+                result = "0"
```
Wrap it into variable

ashokolov commented on app/src/main/java/com/tsu/calculator/MainViewModel.kt:
```diff
@@ -21,154 +14,57 @@ class MainViewModel : ViewModel() {
-        }
+    private fun clear() {
+        model.clear()
+        prepareDisplayValue("0")
```
Use variable

ashokolov commented on app/src/main/java/com/tsu/calculator/MainViewModel.kt:
```diff
@@ -21,154 +14,57 @@ class MainViewModel : ViewModel() {
-        if (result.length > 9) {
-            _display.value = String.format("%.5E", result.toDouble())
+        if (value.length > MainModel.MAX_LENGTH) {
+            _display.value = String.format("%.5E", value.toDouble())
```
Better to let View to decide how to display value.
